### PR TITLE
feat(uniquepaths): add solution for unique paths problem

### PR DESCRIPTION
**Pull request by rares985**
Provides a solution to the "Unique Paths" problem. That is, in how many ways can you traverse an m x n matrix from (0, 0) to (m-1, n-1)

### CODE REVIEW

**Comment by TruongNhanNguyen** [comment]
This looks like a good implementation of the dynamic programming solution for the "Unique Paths" problem. One minor adjustment I would suggest is to remove the `+1` when creating the grid. Since you're using zero-based indexing, allocating `m+1` and `n+1` for the grid creates unnecessary space. Simply use `m` and `n` instead. This adjustment makes the code a bit more concise. Other than that, the logic and implementation seem correct!

File: dynamic/uniquepaths.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// author: Rares Mateizer (https://github.com/rares985)
+package dynamic
+
+// UniquePaths implements the solution to the "Unique Paths" problem
+func UniquePaths(m, n int) int {
+	if m <= 0 || n <= 0 {
+		return 0
+	}
+
+	grid := make([][]int, m+1)
```
Comment (TruongNhanNguyen) [inline dynamic/uniquepaths.go:11]:
```suggestion
	grid := make([][]int, m)
```

File: dynamic/uniquepaths.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	for i := range grid {
+		grid[i] = make([]int, n+1)
```
Comment (TruongNhanNguyen) [inline dynamic/uniquepaths.go:13]:
```suggestion
		grid[i] = make([]int, n)
```

File: dynamic/uniquepaths.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	for i := 0; i < m; i++ {
+		grid[i][0] = 1
+	}
+
+	for j := 0; j < n; j++ {
+		grid[0][j] = 1
+	}
+
+	for i := 1; i <= m; i++ {
+		for j := 1; j <= n; j++ {
```
Comment (TruongNhanNguyen) [inline dynamic/uniquepaths.go:25]:
```suggestion
	for i := 1; i < m; i++ {
		for j := 1; j < n; j++ {
```

**Comment by rares985** [comment]
@yanglbme, @raklaptudirm Could you please review?

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**